### PR TITLE
Map doesn't work with find: Adding a failing test

### PR DESCRIPTION
**Pull request by okdistribute**

### CODE REVIEW

**Comment by okdistribute** [comment]
```
not ok 9 no err
  ---
    operator: notOk
    expected: false
    actual:   {}
    at: DestroyableTransform.<anonymous> (/Users/karissa/dev/node_modules/level-indexer/test.js:74:7)
  ...
[NotFoundError: Key not found in database [mafintosh]]

1..9
```

@mafintosh any idea why this might be happening? I also see the problem with leveldb.

**Comment by okdistribute** [comment]
@mafintosh even when i'm trying to hackingly create the encoder, i still get:

```
mafintosh
{ key: 'name!a0706d6174686961730000!mafintosh',
  value: 'mafintosh' }
not ok 9 no err
  ---
    operator: notOk
    expected: false
    actual:   {}
    at: DestroyableTransform.<anonymous> (/Users/karissa/dev/node_modules/level-indexer/test.js:75:7)
  ...
[NotFoundError: Key not found in database [name!a0706d6166696e746f73680000!mafintosh]]
```

see https://github.com/mafintosh/level-indexer/commit/8adb93c674bb305deb93c139023eeb370c5de698#diff-1dd241c4cd3fd1dd89c570cee98b79ddR65

**Comment by okdistribute** [comment]
it seems like there's some disconnect here as far as encoding & where that gets done. maybe map should be an option & then generated inside so the correct key is looked up?

File: test.js [hunk 1]
```diff
@@ -57,3 +57,24 @@ tape('compound index ranges', function(t) {
...
+
+tape('single index with map', function(t) {
+  var db = memdb()
+  var index = indexer(db, ['name'], {
+    map: function (key, cb) {
+      db.get(index.encode(key) + '!' + key, cb)
+    }
+  })
+  var through = require('through')
+  db.createReadStream().pipe(through(console.log))
```
Comment (okdistribute) [inline test.js:69]: this is super hacky debug code

**Comment by mafintosh** [comment]
@karissa i'm not completely sure I understand your test. Why are you using the encode stuff?

**Comment by okdistribute** [comment]
It doesn't work without it,  either

**Comment by okdistribute** [comment]
Check out  the earlier commit

**Comment by mafintosh** [comment]
Ah ok got it. The map option doesn't work like that. Basically the index only stores primary keys so you can use the map to map that primary key to a value somehow.

``` js
var db = memdb({valueEncoding:'json'})

var map = function(key, cb) {
  db.get(key, cb) // maps primary key to document
}

var index = indexer(indexDb, ['name'], {map:map}) // note indexDb !== db

var doc = {name:'mathias', age:27}
index.add(doc, 'mafintosh', function() { // index the document
  // store the document in another db as well
  // (in most use cases this is done by someone else)
  db.put('mafintosh', doc)
})

...

index.find('mathias', function(err, docs) {
  console.log(docs) // docs are mapped
})
```

**Comment by okdistribute** [comment]
okay cool

**Pull request closed without merging**